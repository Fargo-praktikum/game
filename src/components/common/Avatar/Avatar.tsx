import React, { Fragment } from "react";

import "./Avatar.scss";

import noPhoto from "../../../assets/noPhoto.png";

type AvatarType = {
    avatarSrc?: string
};


const Avatar: React.FC<AvatarType> = ({ avatarSrc }) => {
    console.log("обновился Avatar");
    return (
        <Fragment>
            {!avatarSrc &&
            <img className="avatar" src={noPhoto} alt="фото"/>
            }
            {avatarSrc &&
            <img className="avatar" src={avatarSrc} alt="фото"/>
            }
        </Fragment>
    );
};

export default Avatar;
