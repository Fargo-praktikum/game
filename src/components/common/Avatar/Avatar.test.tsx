import React from "react";

import renderer from "react-test-renderer";
import Avatar from "./Avatar";

describe("Проверка Avatar", () => {
    it("отрисовка без avatarSrc", () => {
        const tree = renderer
            .create(<Avatar/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("отрисовка с avatarSrc", () => {
        const tree = renderer.create(<Avatar
            avatarSrc="https://test.ru"
        />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});



