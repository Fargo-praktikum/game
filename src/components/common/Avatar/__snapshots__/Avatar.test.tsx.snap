// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка Avatar отрисовка без avatarSrc 1`] = `
<img
  alt="фото"
  className="avatar"
  src="test-file-stub"
/>
`;

exports[`Проверка Avatar отрисовка с avatarSrc 1`] = `
<img
  alt="фото"
  className="avatar"
  src="https://test.ru"
/>
`;
